setup.py

```diff
@@ -34,6 +34,7 @@
         'scikit-learn',
         'scipy',
         'tqdm',
+        'urlpath',
```

Comment:
What was the reason to use urlpath? Urlpath has only 25 stars and it's not really maintained.

Reply:
pathlib is not suitable for urls (s3://) so I had to find a different library for handling urls. 
I changed to yarl library: https://github.com/aio-libs/yarl (387 stars)
Hopefully in the future this issue will be maintained within pathlib and we will have an easy way of switching URL to PosixPath

Whi I didn't go with urllib:
The built-in urllib.parse is lacking the functionality we need- we would have to convert the urls to strings and this means the methods that can take either path or url will only use string transformations. So for example it will be impossible to use pathlib's `image_dir.name` to extract the basename. Instead we will have to write `str(image_dir).split('/')[-1]` which to me makes the entire usage or pathlib redundant. The advantage of the urlpath is that it stores urls as urls while providing similar methods as in pathlib (so for example both the URL and Path objects have a method `name`, you can also join further levels of the path simply by adding slash+string: `base / next-level / image_dir` instead of `'/'.join((base, next-level, str(image_dir).split('/')[-1]))`